radis/lbl/factory.py

```diff
@@ -685,6 +692,33 @@ def eq_spectrum(
         # ... calculate broadening  HWHM
         self._calc_broadening_HWHM()
 
+        #################################################
+        # Copied wstep dependent parameters here
+        # calculated range is broader than output waverange to take into account off-range line broadening
+        wavenumber, wavenumber_calc = _generate_wavenumber_range(
+            self.input.wavenum_min,
+            self.input.wavenum_max,
+            self.params.wstep,
+            self.params.broadening_max_width,
+        )
+        wbroad_centered = _generate_broadening_range(
+            self.params.wstep, self.params.broadening_max_width
+        )
+
+        # Get boolean array that extracts the reduced range `wavenumber` from `wavenumber_calc`
+        woutrange = np.in1d(wavenumber_calc, wavenumber, assume_unique=True)
+        self.wbroad_centered = wbroad_centered
+        self.wavenumber = wavenumber
+        self.wavenumber_calc = wavenumber_calc
+        self.woutrange = woutrange
+        self.params.wavenum_min_calc = wavenumber_calc[0]
+        self.params.wavenum_max_calc = wavenumber_calc[-1]
+        print("l ->wavenum - ", self.wavenumber)
+        print("l ->wavenum_calc - ", self.wavenumber_calc)
+        print("l ->wbroad - ", self.wbroad_centered)
+
+        #####################################################
+
```

Comment:
can you move this in a function ? It's probably the right place to compute wstep_auto too.